packages/admin/cms-admin/src/generator/future/generateGrid/generateGridToolbar.ts

```diff
@@ -100,11 +95,23 @@ const renderMoreActionsMenu = (excelExport: boolean | undefined) => {
     />`;
 };
 
-const renderToolbarActions = (forwardToolbarAction: boolean | undefined, addItemText: string) => {
+const renderToolbarActions = (
+    forwardToolbarAction: boolean | undefined,
+    instanceGqlType: string,
+    gqlType: string,
+    newEntryText: string | undefined,
+) => {
     if (forwardToolbarAction) {
         return `{toolbarAction && <ToolbarActions>{toolbarAction}</ToolbarActions>}`;
     }
 
+    let addItemText = getFormattedMessageNode(`${instanceGqlType}.new${gqlType}`, `New ${camelCaseToHumanReadable(gqlType)}`);
+
+    if (newEntryText) {
+        // This requires a different message-id to prevent other grids with this type but no custom text from having the same id
```

Comment:
how do we solve this issue for other cases?

Reply:
I don't think we do but I'm not sure 😅 
@johnnyomair do you have an idea?

Reply:
I don't think we've had this use case in the generator up until now. What do you think about re-using (and misusing) the fragment name for this? This must be unique across the whole project.

Reply:
Seems like a good solution for now 😄

Reply:
Done in a8c7dee4064aeea58886ddd8b524f0fcfa593ba1

> Make message-id of add button unique for each grid using it's unique fragment